webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -106,22 +107,18 @@ function ChannelSettingsAccessRulesTab({
         };
 
         loadAttributes();
-    }, [actions]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    // Load existing channel access rules
+    // Load existing channel access rules (only when channel changes)
     useEffect(() => {
         const loadChannelPolicy = async () => {
             try {
                 const result = await actions.getChannelPolicy(channel.id);
                 if (result.data) {
                     // Extract expression from the policy rules
                     const existingExpression = result.data.rules?.[0]?.expression || '';
-                    let existingAutoSync = result.data.active || false;
-
-                    // If system policies force auto-sync, override the channel setting
-                    if (systemPolicyForcesAutoSync) {
-                        existingAutoSync = true;
-                    }
```

Comment:
This (and the next change) look like a "loss of functionality", and I don't see this functionality being recovered. Do you mind explaining why this is no longer needed?

Reply:
Do we have already tests that verify this logic is correct? If not, it would be great to add them.